src/main.go

```diff
@@ -162,10 +163,44 @@ func setSessionVariable(db *sql.DB) {
 	if _, err := db.Exec("SET @@tidb_enable_pseudo_for_outdated_stats=false"); err != nil {
 		log.Fatalf("Executing \"SET @@tidb_enable_pseudo_for_outdated_stats=false\" err[%v]", err)
 	}
-	// enable tidb_enable_analyze_snapshot in order to let analyze request with SI isolation level to get accurate response
-	if _, err := db.Exec("SET @@tidb_enable_analyze_snapshot=1"); err != nil {
-		log.Fatalf("Executing \"SET @@tidb_enable_analyze_snapshot=1\" err[%v]", err)
+	rows, err := db.Query("select version from INFORMATION_SCHEMA.CLUSTER_INFO where type = 'tidb' limit 1;")
+	if err != nil {
+		log.Fatalf("Executing \"select version from INFORMATION_SCHEMA.CLUSTER_INFO where type = 'tidb' limit 1;\" err[%v]", err)
+	}
+	defer rows.Close()
+	var version string
+	for rows.Next() {
+		err := rows.Scan(&version)
+		if err != nil {
+			log.Fatalf("Executing \"select version from INFORMATION_SCHEMA.CLUSTER_INFO where type = 'tidb' limit 1;\" err[%v]", err)
+		}
+	}
+	isVersionAble, err := checkVersion(version)
```

Comment:
Could you rename it to something like `isSupportedVersion`?

Reply:
updated.